Bugfix: disable channels after IO if multiple tca9548a  I2C multiplexers are configured

kahrendt opened the pull request:
# What does this implement/fix?

If there are multiple tca9548a I2C multiplexers configured on the same bus, then there are potential address conflicts. If you have two components with the same address attached to two tca9548a multiplexers, they will both attempt to communicate at the same time. This PR disables all the channels on a tca9548a after any IO if there is more than one multiplexer configured.

I removed the MULTI_CONF option in ``__init__.py``, and used a `cv.ensure_list` to determine if more than tca9548a is configured. If only one tca9548a is configured, then the  component does not disable all channels and matches the current behavior.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** not applicable

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** not applicable

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

tca9548a:
  - address: 0x70
    id: multiplex0
    i2c_id: direct_bus
    channels:
      - bus_id: multplex0channel0
        channel: 0
  - address: 0x71
    id: multiplex1    
    i2c_id: direct_bus
    channels:
      - bus_id: multplex1channel0
        channel: 0

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

kahrendt committed:
add function to disable channels

kahrendt committed:
clang-tidy fix

jesserockz reviewed (changes requested):
I dont think we need to care if there are multiple multiplexers. Just disable the channel always after a transaction.

Also don't mark here as failed, just log an error and `set_status_error` perhaps.

kahrendt committed:
always disable channels after IO

kahrendt commented:
Sounds good, thanks for the feedback! I reverted back to the original ``__init__.py`` with ``MULTI_CONF=True``. The component now disables all channels after any communication. I removed the ``current_channel_`` variable and checks, as the channel will need to be written every time if things are working correctly.
